src/feature/hs/hs_intropoint.c

```diff
@@ -480,6 +484,15 @@ handle_introduce1(or_circuit_t *client_circ, const uint8_t *request,
     }
   }
 
+  /* Before sending, lets make sure this cell can be sent on the service
+   * circuit asking the DoS defenses. */
+  if (!hs_dos_can_send_intro2(service_circ)) {
+    log_info(LD_PROTOCOL, "Can't relay INTRODUCE1 v3 cell due to DoS "
```

Comment:
Perhaps this log needs ratelimiting since it's gonna hit hard if it hits.

Reply:
oh indeed!

Fixup ebb887afb9

Reply:
LGTM!